services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/MetersSpec.groovy

```diff
@@ -250,16 +260,132 @@ class MetersSpec extends BaseSpecification {
         newMeters*.rate.every { it == flowRate }
 
         and: "New meters burst size should be 1024 kbit/s regardless the flow speed"
-        newMeters*.burstSize.every { it == burstSize }
+        newMeters*.burstSize.every { it == expectedBurstSize }
 
         cleanup: "Delete the flow"
         flowHelper.deleteFlow(flow.id)
 
         where:
-        flowRate | burstSize
-        975      | 1024 //should have been 1023, but use the minimum 1024
-        1000     | 1050 //the 1.05 multiplier
-        30478    | 32000 //should have been 32001, but use the maximum 32000
+        flowRate << [
+            //flowRate below min
+            ((CENTEC_MIN_BURST - 1) / getBurstCoef()).toBigDecimal().setScale(0, RoundingMode.CEILING).toLong(),
+            1000, //casual middle value
+            //flowRate above max
+            ((CENTEC_MAX_BURST + 1) / getBurstCoef()).toBigDecimal().setScale(0, RoundingMode.CEILING).toLong()
+        ]
+        sw = getCentecSwitches()[0]
+        expectedBurstSize = getExpectedBurst(sw.dpId, flowRate)
+    }
+
+    @Unroll
+    def "System allows to reset meter values to defaults without reinstalling rules for #data.description flow"() {
+        given: "A flow with custom meter rate and burst, that differ from defaults"
+        def src = data.srcSwitch
+        def dst = data.dstSwitch
+        assumeTrue("Desired switch combination is not available in current topology", src && dst)
+        def flow = flowHelper.randomFlow(src, dst)
+        flow.maximumBandwidth = 1000
+        flowHelper.addFlow(flow)
+        /*at this point meters are set for given flow. Now update flow bandwidth directly via DB, so that existing meter
+        rate and burst is no longer correspond to the flow bandwidth*/
+        def newBandwidth = 2000
+        database.updateFlowBandwidth(flow.id, newBandwidth)
+        //at this point existing meters do not correspond with the flow
+        //now save some original data for further comparison before resetting meters
+        Map<SwitchId, SwitchFlowEntries> originalRules = [src.dpId, dst.dpId].collectEntries {
+            [(it): northbound.getSwitchRules(it)]
+        }
+        Map<SwitchId, List<MeterEntry>> originalMeters = [src.dpId, dst.dpId].collectEntries {
+            [(it): northbound.getAllMeters(it).meterEntries]
+        }
+        def defaultMeters = { it.meterId < Constants.MAX_DEFAULT_METER_ID }
+        def flowMeters = { it.meterId >= Constants.MAX_DEFAULT_METER_ID }
+
+        when: "Ask system to reset meters for the flow"
+        def response = northbound.resetMeters(flow.id)
+
+        then: "Response contains correct info about new meter values"
+        [response.srcMeter, response.dstMeter].each { switchMeterEntries ->
```

Comment:
So, essentially resetMeters(flow.id) updates flow's bandwidth on switches (synchronizing it with the value in the database which was updated previously). Would be good to mention it in a comment or in the step description, as this is a tricky part.

Reply:
it is not updating flow's bandwidth. Flow bandwidth itself is updated during database operation. ResetMeters updates meter rate and meter burst to match the flow bandwidth